README.md

```diff
@@ -5,4 +5,402 @@
 [![Build Status](https://api.travis-ci.com/pingcap/failpoint.svg?branch=master)](https://travis-ci.com/pingcap/failpoint)
 [![Coverage Status](https://codecov.io/gh/pingcap/failpoint/branch/master/graph/badge.svg)](https://codecov.io/gh/pingcap/failpoint)
 
-Golang Failepoint
+An implementation of [failpoints][failpoint] for golang.
+
+[failpoint]: http://www.freebsd.org/cgi/man.cgi?query=fail
+
+## Features
+
+- Define failpoint in valid Golang code, not comments or anything else
+- Failpoint does not have any extra cost
+
+    - Will not take effect on regular logic
+    - Will not cause regular code performance regression
+    - Failpoint code will not appear in the final binary
+
+- Failpoint routine is easy to write/read and checked by a compiler
```

Comment:
Failpoint routine is easy to write/read and **should be** checked by a compiler?